fvm/evm/stdlib/contract.go

```diff
@@ -38,6 +42,491 @@ var evmAddressBytesType = sema.NewConstantSizedType(nil, sema.UInt8Type, types.A
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int8Value:
+					values = append(values, int8(value))
+					typ, err := gethABI.NewType("int8", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int16Value:
+					values = append(values, int16(value))
+					typ, err := gethABI.NewType("int16", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int32Value:
+					values = append(values, int32(value))
+					typ, err := gethABI.NewType("int32", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int64Value:
+					values = append(values, int64(value))
+					typ, err := gethABI.NewType("int64", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int128Value:
+					values = append(values, value.BigInt)
+					typ, err := gethABI.NewType("int128", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.Int256Value:
+					values = append(values, value.BigInt)
+					typ, err := gethABI.NewType("int256", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case interpreter.BoolValue:
+					values = append(values, bool(value))
+					typ, err := gethABI.NewType("bool", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				case *interpreter.CompositeValue:
+					if value.QualifiedIdentifier == "EVM.EVMAddress" {
+						bytes, err := interpreter.ByteArrayValueToByteSlice(
+							inter,
+							value.GetMember(inter, locationRange, evmAddressTypeBytesFieldName),
+							locationRange,
+						)
+						if err != nil {
+							panic(err)
+						}
+						values = append(values, gethCommon.Address(bytes))
+						typ, err := gethABI.NewType("address", "", nil)
+						if err != nil {
+							panic(err)
+						}
+						arguments = append(arguments, gethABI.Argument{Type: typ})
+					}
+				case *interpreter.ArrayValue:
+					elements := make([]uint64, 0)
+					value.Iterate(inter, func(element interpreter.Value) (resume bool) {
+						v, ok := element.(interpreter.UInt64Value)
+						if !ok {
+							panic(
+								fmt.Errorf(
+									"unsupported array value type: %v",
+									element.StaticType(inter),
+								),
+							)
+						}
+						elements = append(elements, uint64(v))
+
+						return true
+					})
+					values = append(values, elements)
+					typ, err := gethABI.NewType("uint64[]", "", nil)
+					if err != nil {
+						panic(err)
+					}
+					arguments = append(arguments, gethABI.Argument{Type: typ})
+				}
```

Comment:
Add a default case and panic with a new user error that indicates the passed value is not supported / cannot be encoded. Currently the switch is silently ignoring unsupported values. Maybe also add test cases